docs/developer/decision-records/2022-10-10-naming-conventions/README.md

```diff
@@ -0,0 +1,122 @@
+such a violation:
+
+```kotlin
+// settings.gradle.kts
+include(":core:common:transfer-process")
+include(":extensions:sql:transfer-process")
+```
+
+The EDC project has checks in place to make sure module IDs are unique.
+
+> Rule 1: Modules must have unique names.
+
+In addition, the _module name_ should give a hint what is in the module, without being too verbose. The
+earlier example would be a bad one, because "transfer-process" does not indicate what the contents could
+be. This is especially important because we require the Maven's _artifactId_ to be equal to module names.
+
+Here are some bad examples:
+- `:core:common:transfer-process:validation`: bad because "validation" is likely to be not unique and isolated it only indicates that it has to do with validation, but not in _what context_.
+- `:core:dataplane:framework`: again, "framework" is liable to cause conflicts, and in addition, it's a very generic, unspecific term
+
+Refactoring these bad examples, we could arrive at these:
+- `:core:common:transfer-process:transfer-process-validation`: could contain validation functions
+- `:core:dataplane:dataplane-framework`: would contain default impls and platform code for the dataplane
+
+> Rule 2: Module names should indicate what the contents are.
+
+> Rule 3: Module names must be identical to the Maven artifactId (if one is published).
+
+### Maven artifactId
+
+The EDC project uses the same `groupId = "org.eclipse.edc"` across all sub-projects, which means all
+_artifactIds_ must be unique across multiple software components to avoid conflicts.
+
+> Rule 4: A Maven artifactId must be unique within the groupId.
+
+> Rule 5: A Maven artifactId must be identical to the module name (cf. [Rule 3](#gradle-module-name)).
+
+### Java package name
+
+Following Oracle's [Java package naming conventions](https://docs.oracle.com/javase/tutorial/java/package/namingpkgs.html),
+EDC's base package name is `org.eclipse.edc`, followed by domain and then function. For example, the
+Policy SPI should be located at `org.eclipse.edc.policy.spi` rather than `org.eclipse.edc.spi.policy`.
+Here, `policy` is the "domain", i.e. the thematic context, and `spi` would be the "function", i.e. what
+kind of package it is or what purpose it serves.
+
+The module name could be a helpful reference for the package name, replacing dashes with dots.
+
+Other positive examples would be:
+- `org.eclipse.edc.transferprocess.validation`: domain is `transferprocess`, and it's a package that deals with validation
+- `org.eclipse.edc.dataplane.framework.manager`: here the domain is `dataplane.framework`, so all code should be beneath that directory.
+
+This helps to avoid split packages.
+
+> Rule 6: Package names should first contain the domain, and then the function.
+
+### Configuration properties
+
+Configuration properties should have a unique prefix `edc` to avoid clashes, when EDC gets embedded
```

Comment:
I remember we decided to remove the `edc` prefix as it was unnecessary, using the module name as prefix instead. I am correct?

Reply:
Fine with removing, I had a look, I noted it down as decision. With the module name, I am not sure.. this could get ugly :D However, this would be simple and self-descriptive

Reply:
I have no strong opinion on this, I just remember we had a different decision.
Just for confirm: @paullatzelsperger @jimmarino @ronjaquensel are we ok with the `edc` prefix for config settings? Wasn't it considered implicit?

Reply:
@ndr-brt I also think we decided to omit this prefix, but thinking that the EDC may be embedded in/combined with other frameworks where potentially configuration might get mixed, I think it makes sense to add the prefix. But I also don't have a strong opinion on this.

Reply:
if nobody has a strong optinion, I would merge this as-is